evalml/automl/callbacks.py

```diff
@@ -44,4 +44,6 @@ def log_error_callback(exception, traceback, automl, **kwargs):
         logger.info(f"\t\t\tFold {fold_num}: All scores will be replaced with nan.")
     logger.info(f"\t\t\tFold {fold_num}: Please check {logger.handlers[1].baseFilename} for the current hyperparameters and stack trace.")
     logger.debug(f"\t\t\tFold {fold_num}: Hyperparameters:\n\t{pipeline.hyperparameters}")
+    trace = '\n'.join(traceback)
```

Comment:
If the exception a `PipelineScoreError`, then `e.message` contains information about how each objective failed. I think we should add that to the traceback. 

Maybe we can do something like this:

```python
    trace = '\n'.join(traceback)
    if isinstance(exception, PipelineScoreError):
        logger.info(f"\t\t\tFold {fold_num}: Encountered an error scoring the following objectives: {', '.join(exception.exceptions)}.")
        logger.info(f"\t\t\tFold {fold_num}: The scores for these objectives will be replaced with nan.")
        trace += e.message
    else:
        logger.info(f"\t\t\tFold {fold_num}: Encountered an error.")
        logger.info(f"\t\t\tFold {fold_num}: All scores will be replaced with nan.")

    logger.info(f"\t\t\tFold {fold_num}: Please check {logger.handlers[1].baseFilename} for the current hyperparameters and stack trace.")
    logger.info(f"\t\t\tFold {fold_num}: Exception during automl search: {str(exception.__name__)}")
    logger.debug(f"\t\t\tFold {fold_num}: Hyperparameters:\n\t{pipeline.hyperparameters}")
    logger.debug(f"\t\t\tFold {fold_num}: Traceback:\n{trace}")
```